src/hotspot/share/prims/scopedMemoryAccess.cpp

```diff
@@ -138,17 +159,43 @@ class CloseScopedMemoryClosure : public HandshakeClosure {
 };
 
 /*
- * This function issues a global handshake operation with all
- * Java threads. This is useful for implementing asymmetric
- * dekker synchronization schemes, where expensive synchronization
- * in performance sensitive common paths, may be shifted to
- * a less common slow path instead.
- * Top frames containg obj will be deoptimized.
+ * This functin performs a thread-local handshake against all threads running at the time
+ * the given scope (deopt) was closed. If the handshake closure finds that a thread has
+ * safepointed inside a scoped method (that is, a method inside the ScopedMemoryAccess class
+ * annotated with the '@Scoped' annotation), whose local variables mention the scope being closed
+ * (deopt), the thread is added to a problematic stack. After the handshake, each thread in
+ * the problematic stack is handshaked again, individually, to check that it has exited
```

Comment:
I suggest saying "list" rather  than "stack" as VM developers tend to assume "stack" means thread stack.